pkg/kubectl/resource_printer.go

```diff
@@ -1505,6 +1508,34 @@ func getNodeExternalIP(node *api.Node) string {
 	return "<none>"
 }
 
+// findNodeRole returns the role of a given node, or "" if none found.
+// The role is determined by looking in order for:
+// * a kubernetes.io/role label
+// * a kubeadm.alpha.kubernetes.io/role label
+// * a taint with Key 'dedicated'
+// If no role is found, ("") is returned
+func findNodeRole(node *api.Node) string {
+	if role := node.Labels["kubernetes.io/role"]; role != "" {
```

Comment:
If these are official, they need to be constants.  I can't find mention of this in HEAD, though it looks like someone is using it as a prefix for other annotation in the aws provider only.